controller/aks-cluster-config-handler.go

```diff
@@ -843,18 +843,19 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, config *aksv1.
 			clusterSpecCopy.ResourceGroup = config.Spec.ResourceGroup
 			clusterSpecCopy.ResourceLocation = config.Spec.ResourceLocation
 			clusterSpecCopy.ClusterName = config.Spec.ClusterName
-		}
-		clusterSpecCopy.NodePools = make([]aksv1.AKSNodePool, 0, len(config.Spec.NodePools))
-		for _, n := range config.Spec.NodePools {
-			if _, ok := upstreamNodePools[*n.Name]; ok {
-				clusterSpecCopy.NodePools = append(clusterSpecCopy.NodePools, n)
```

Comment:
Why are you changing logic for provisioned clusters by Rancher? In results of this change non-imported clusters won't be updated.

Reply:
The loop iterating over config.NodePools checks if that NodePool is present in upstreamNodePools, which means that no new pool can be added or deleted. This logic seems confusing. Changing this only for Imported cluster seems logical

Reply:
Updates for NodePools need to work for both rancher managed clusters and imported one. Your change will break this and only imported cluster will be updated.